src/allreduce_base.cc

```diff
@@ -211,6 +228,15 @@ void AllreduceBase::SetParam(const char *name, const char *val) {
       throw std::runtime_error("invalid value of DMLC_WORKER_STOP_PROCESS_ON_ERROR");
     }
   }
+  if (!strcmp(name, "rabit_cache")) {
+    rabit_cache = atoi(val);
```

Comment:
we do check if(rabit_bootstrap_cache) so if it's negative,user still opt-in this feature
only when rabit_bootstrap_cache default (0), it will not insert/fetch bootstrapcache